config.json

```diff
@@ -2,5 +2,6 @@
     "deep_filter_path": "MediaProcessor/res/deep-filter-0.5.6-x86_64-unknown-linux-musl",
     "downloads_dir": "downloads",
     "ffmpeg_path": "/usr/bin/ffmpeg",
-    "upload_folder": "uploads" 
+    "upload_folder": "uploads",
+    "num_chunks": 0
```

Comment:
Please rename it to `num_threads`. Users should not be concerned with how we manage our chunking wrt thread availability, and we may change how we chunk later on, and that shouldn't break the config or the user API. 

Also, it shouldn't default to 1 and not 0.

Reply:
> Also, it shouldn't default to 1 and not 0.
I think we shouldn't place the num_threads in the config at starting. Beacuse most of the users are not concerned about the threads it uses.
so there is no point of default value.
users can add this setting if they want to limit the number of threads.
???

Reply:
You're right. We should then introduce a new field in the config `limit_thread_count` that defaults to `false` and rename `num_chunks` to `max_thread_count` which you can default to a low number, e.g. 4. 

`max_thread_count` is only relevant if `limit_thread_count` is `true`.